lib/srv/desktop/tdp/conn.go

```diff
@@ -29,6 +31,12 @@ import (
 type Conn struct {
 	rw   io.ReadWriter
 	bufr *bufio.Reader
+	once sync.Once
```

Comment:
Maybe:

```suggestion
	closeOnce sync.Once
```

or add a comment explaining what the `sync.Once` is used for.